Apps/RaptureIntTests/src/test/java/rapture/httpapi/blob/BlobApiTests.java

```diff
@@ -0,0 +1,417 @@
+        Assert.assertTrue (metaDataMap.containsKey("modifiedTimestamp"));
+    }
+    
+    @Test (groups={"blob","mongo", "smoke"},enabled=true, expectedExceptions=RaptureException.class,dataProvider = "configTypes")
+    public void testNullBlobContents(String dbType, String configType) {
+       
+        String blobAuthority="test.blobauthority";
+        String metaAuthority="metatest.authority";
+
+        String blobURI = RaptureURI.builder(BLOB, blobAuthority+System.nanoTime()).asString();
+        String CONFIG = String.format(mongoRepoConfigTemplate, configType, blobAuthority);
+        String CONFIGMETA = String.format(mongoRepoMetaConfigTemplate,metaAuthority);
+        System.out.println("Blob Config String: " + CONFIG);
+        try {
+            blobApi.createBlobRepo(blobURI,CONFIG,CONFIGMETA);
+        }
+        catch (Exception e) {
+            Assert.fail();
+        }
+               
+        blobApi.putBlob(blobURI,null, "application/text");
+        
+    }
+    
+    @Test (groups={"blob","mongo", "smoke"},dataProvider = "configTypes")
+    public void testBlobAppend (String dbType,String configType){
+        String blobAuthority="test.blobauthority";
+        String metaAuthority="meta.blobauthority";
+        long maxContentSize=10000L;
+        
+        Random rand = new Random();
+
+        String blobURI=RaptureURI.builder(BLOB, blobAuthority+System.nanoTime()).asString();
+       
+
+        String CONFIG = String.format(mongoRepoConfigTemplate, configType, blobAuthority);
+        String CONFIGMETA = String.format(mongoRepoMetaConfigTemplate,metaAuthority);
+   
+        try {
+            blobApi.createBlobRepo(blobURI,CONFIG,CONFIGMETA);
+        }
+        catch (Exception e) {
+            Assert.fail();
+        }
+              
+        long content_size=Math.abs(rand.nextLong() % maxContentSize);
+        String currBlobURI=blobURI + Thread.currentThread().getId() + "_" + content_size + "_" + System.nanoTime();
+        System.out.println ("Creating blob with content size="+content_size);
+        System.out.println ("URI="+currBlobURI);
+        String currContent="INITIAL CONTENT";
+
+        System.out.println("storing and appending to blob: " + currBlobURI);
+        try{
+            blobApi.putBlob(currBlobURI,currContent.getBytes(), "application/text"); 
+        } catch (Exception e) {
+            System.out.println("Exception thrown: " + e);
+        }
+        
+        blobApi.addBlobContent(currBlobURI, "MORE CONTENT".getBytes());
+        Assert.assertEquals (new String (blobApi.getBlob(currBlobURI).getContent()),currContent+"MORE CONTENT");
+        
+        blobApi.addBlobContent(currBlobURI, "EVEN MORE CONTENT".getBytes());
+        Assert.assertEquals (new String (blobApi.getBlob(currBlobURI).getContent()),currContent+"MORE CONTENT"+"EVEN MORE CONTENT");
+    }
+    
+    
+    @Test (groups={"blob","mongo", "smoke"},dataProvider = "configTypes")
```

Comment:
what is the design for the groups? how are you handling running with fts off/on? are you coordinating groups with Dave?